src/Control/Monad/Eff/Random.purs

```diff
@@ -28,7 +28,18 @@ foreign import random :: forall e. Eff (random :: RANDOM | e) Number
 randomInt :: forall e. Int -> Int -> Eff (random :: RANDOM | e) Int
 randomInt low high = do
   n <- random
-  pure <<< U.fromJust <<< fromNumber <<< Math.floor $ toNumber (high - low + one) * n + toNumber low
+  let asNumber = (toNumber high - toNumber low + one) * n + toNumber low
+  return $ safeFromNumber asNumber
+
+-- | Safely convert a Number into an Int, by first flooring and then clamping
+-- to the valid integer range.
+safeFromNumber :: Number -> Int
```

Comment:
Ah, just saw your second comment!

Regardless of what we do about different backends, `top :: Int` should still work on any backend, right? Perhaps Prelude can get the maximum and minimum values for the `Bounded Int` instance from FFI files, so that we can make them differ across backends?